src/components/calcite-pick-list/shared-list-tests.ts

```diff
@@ -17,7 +17,7 @@ export const tests = {
 
         await item1.click();
         await item2.click();
-        expect(toggleSpy).toHaveReceivedEventTimes(2);
+        expect(toggleSpy).toHaveReceivedEventTimes(3);
```

Comment:
This is still one time too many. The event should trigger once for every click. 2 Clicks should be 2 events emitted.

Reply:
The event is called one more time since it is called when selected is set to false now due to `selected` being the source of truth for that item event.

Previously, setting selected to false on an item didn't trigger an event. Now it does, hence the 1 more event firing. Users should know when something is set to false as well.

Reply:
Hmm. I'm concerned this may cause bugs. If you're reacting to the list changing - you're going to get the same list of selected items twice in this scenario.
I'd expect the `calciteListItemChange` event to fire 3 times (once for selecting item one, once for deselecting item one, and again for selecting item 2).
But the `calciteListChange` event I'd expect to fire twice (one for when item one is the selected item, and once for when item two is the selected item).

Reply:
> I'm concerned this may cause bugs. If you're reacting to the list changing - you're going to get the same list of selected items twice in this scenario.

Logically, it shouldn't cause any bugs. Can we not fire the event if the same selected item list exists? That logic would need to be added.

> 2). But the calciteListChange event I'd expect to fire twice (one for when item one is the selected item, and once for when item two is the selected item).

Seems like some kind of throttle would have to happen for this?

Reply:
We could short-circuit the event from firing if the list didn't change but we'd need to start storing the previous state - could be error prone. It's only needed when utilizing the mutable prop method - that's why I avoided that. Depending on what the callback does - it could cause errors in the app, like if it were incrementing some value it would be off. 

Have you given any consideration toward the simpler approach of keeping the prop and state split and changing the name of the prop reflecting the state? Would be a 1-line change.

Reply:
> Have you given any consideration toward the simpler approach of keeping the prop and state split and changing the name of the prop reflecting the state? Would be a 1-line change.

If we did that, `isSelected={this.isSelected}`, then `isSelected` would have to be undocumented as a `@Prop` otherwise it would be the same issue. I don't think we should go that route because of that.

Reply:
So, we can either...

1. Be OK with the additional event
2. Not emit the event if there is no change.

Reply:
@jcfranco can you chime in?

Reply:
Fixed: not emitting event when no selection occurs and its not using multiple.